### PR TITLE
[ADD]business_requirement_deliverable_8.0-12206-add-new-module-busine…

### PR DESCRIPTION
**Pull request by cialuo**
[ADD]business_requirement_deliverable_8.0-12206-add-new-module-busine…
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/xie8899/project/pull/15%23issuecomment-191676081%22%2C%20%22https%3A//github.com/xie8899/project/pull/15%23discussion_r54859417%22%2C%20%22https%3A//github.com/xie8899/project/pull/15%23discussion_r54859493%22%2C%20%22https%3A//github.com/xie8899/project/pull/15%23discussion_r54859522%22%2C%20%22https%3A//github.com/xie8899/project/pull/15%23discussion_r54859688%22%2C%20%22https%3A//github.com/xie8899/project/pull/15%23discussion_r54859716%22%2C%20%22https%3A//github.com/xie8899/project/pull/15%23issuecomment-191714713%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/xie8899/project/pull/15%23issuecomment-191676081%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%40cialuo%20%20Please%20fix%20conflicts%2C%20thanks.%22%2C%20%22created_at%22%3A%20%222016-03-03T09%3A26%3A30Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3785252%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/duanyp1991%22%7D%7D%2C%20%7B%22body%22%3A%20%22%2316%20%22%2C%20%22created_at%22%3A%20%222016-03-03T11%3A23%3A04Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7767792%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/cialuo%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%201b042f16c7e25dc1b3258fb853141e4a9fffc4e0%20business_requirement_deliverable_cost/tests/test_br.py%2019%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/xie8899/project/pull/15%23discussion_r54859716%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22remove%20this%22%2C%20%22created_at%22%3A%20%222016-03-03T10%3A12%3A07Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3785252%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/duanyp1991%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20business_requirement_deliverable_cost/tests/test_br.py%3AL1-162%22%7D%2C%20%22Pull%201b042f16c7e25dc1b3258fb853141e4a9fffc4e0%20business_requirement_deliverable_cost/models/business.py%20131%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/xie8899/project/pull/15%23discussion_r54859688%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22same%20here%22%2C%20%22created_at%22%3A%20%222016-03-03T10%3A11%3A55Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3785252%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/duanyp1991%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20business_requirement_deliverable_cost/models/business.py%3AL1-139%22%7D%2C%20%22Pull%201b042f16c7e25dc1b3258fb853141e4a9fffc4e0%20business_requirement_deliverable_cost/__openerp__.py%207%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/xie8899/project/pull/15%23discussion_r54859417%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22version%20should%20be%208%22%2C%20%22created_at%22%3A%20%222016-03-03T10%3A09%3A33Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3785252%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/duanyp1991%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20business_requirement_deliverable_cost/__openerp__.py%3AL1-22%22%7D%2C%20%22Pull%201b042f16c7e25dc1b3258fb853141e4a9fffc4e0%20business_requirement_deliverable_cost/models/business.py%20119%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/xie8899/project/pull/15%23discussion_r54859522%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22same%20here%22%2C%20%22created_at%22%3A%20%222016-03-03T10%3A10%3A32Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3785252%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/duanyp1991%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20business_requirement_deliverable_cost/models/business.py%3AL1-139%22%7D%2C%20%22Pull%201b042f16c7e25dc1b3258fb853141e4a9fffc4e0%20business_requirement_deliverable_cost/models/business.py%20107%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/xie8899/project/pull/15%23discussion_r54859493%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22dont%20use%20%60api.one%60.%20it%27s%20not%20recomanded.%22%2C%20%22created_at%22%3A%20%222016-03-03T10%3A10%3A20Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3785252%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/duanyp1991%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20business_requirement_deliverable_cost/models/business.py%3AL1-139%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/xie8899/project/pull/15#issuecomment-191676081'>General Comment</a></b>
- <a href='https://github.com/duanyp1991'></a> @cialuo  Please fix conflicts, thanks.
- <a href='https://github.com/cialuo'></a> #16
- [ ] <a href='#crh-comment-Pull 1b042f16c7e25dc1b3258fb853141e4a9fffc4e0 business_requirement_deliverable_cost/__openerp__.py 7'></a> &nbsp;<b><a href='https://github.com/xie8899/project/pull/15#discussion_r54859417'>File: business_requirement_deliverable_cost/**openerp**.py:L1-22</a></b>
- <a href='https://github.com/duanyp1991'></a> version should be 8
- [ ] <a href='#crh-comment-Pull 1b042f16c7e25dc1b3258fb853141e4a9fffc4e0 business_requirement_deliverable_cost/models/business.py 107'></a> &nbsp;<b><a href='https://github.com/xie8899/project/pull/15#discussion_r54859493'>File: business_requirement_deliverable_cost/models/business.py:L1-139</a></b>
- <a href='https://github.com/duanyp1991'></a> dont use `api.one`. it's not recomanded.
- [ ] <a href='#crh-comment-Pull 1b042f16c7e25dc1b3258fb853141e4a9fffc4e0 business_requirement_deliverable_cost/models/business.py 119'></a> &nbsp;<b><a href='https://github.com/xie8899/project/pull/15#discussion_r54859522'>File: business_requirement_deliverable_cost/models/business.py:L1-139</a></b>
- <a href='https://github.com/duanyp1991'></a> same here
- [ ] <a href='#crh-comment-Pull 1b042f16c7e25dc1b3258fb853141e4a9fffc4e0 business_requirement_deliverable_cost/models/business.py 131'></a> &nbsp;<b><a href='https://github.com/xie8899/project/pull/15#discussion_r54859688'>File: business_requirement_deliverable_cost/models/business.py:L1-139</a></b>
- <a href='https://github.com/duanyp1991'></a> same here
- [ ] <a href='#crh-comment-Pull 1b042f16c7e25dc1b3258fb853141e4a9fffc4e0 business_requirement_deliverable_cost/tests/test_br.py 19'></a> &nbsp;<b><a href='https://github.com/xie8899/project/pull/15#discussion_r54859716'>File: business_requirement_deliverable_cost/tests/test_br.py:L1-162</a></b>
- <a href='https://github.com/duanyp1991'></a> remove this

<a href='https://www.codereviewhub.com/xie8899/project/pull/15?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/xie8899/project/pull/15?approve=1'></a>&nbsp;<a href='https://github.com/xie8899/project/pull/15'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by duanyp1991** [comment]
@cialuo  Please fix conflicts, thanks.

File: business_requirement_deliverable_cost/__openerp__.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# -*- coding: utf-8 -*-
+# © 2016 Elico Corp (www.elico-corp.com).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+{
+    "name": "Business Qequirement Deliverable Cost Control",
+    "summary": "Business Qequirement Deliverable Cost Control Module Project",
+    "version": "9.0.1.0.1",
```
Comment (duanyp1991) [inline business_requirement_deliverable_cost/__openerp__.py:7]: version should be 8

File: business_requirement_deliverable_cost/models/business.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+group_business_requirement_cost_control',
+    )
+    gross_profit = fields.Float(
+        string='Estimated Gross Profit',
+        compute='_compute_gross_profit',
+        groups='business_requirement_deliverable_cost.\
+group_business_requirement_cost_control',
+    )
+
+    @api.one
```
Comment (duanyp1991) [inline business_requirement_deliverable_cost/models/business.py:107]: dont use `api.one`. it's not recomanded.

File: business_requirement_deliverable_cost/models/business.py [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    )
+    def _compute_resource_tasks_total(self):
+        if self.deliverable_lines:
+            self.resource_tasks_total = sum(
+                self.mapped('deliverable_lines').mapped(
+                    'resource_ids').filtered(
+                    lambda r: r.resource_type == 'task').mapped('price_total')
+            )
+
+    @api.one
```
Comment (duanyp1991) [inline business_requirement_deliverable_cost/models/business.py:119]: same here

File: business_requirement_deliverable_cost/models/business.py [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+    )
+    def _compute_resource_procurement_total(self):
+        if self.deliverable_lines:
+            self.resource_procurement_total = sum(
+                self.mapped('deliverable_lines').mapped(
+                    'resource_ids').filtered(
+                    lambda r: r.resource_type == 'procurement').mapped(
+                    'price_total'))
+
+    @api.one
```
Comment (duanyp1991) [inline business_requirement_deliverable_cost/models/business.py:131]: same here

File: business_requirement_deliverable_cost/tests/test_br.py [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+        super(BusinessRequirementTestCase, self).setUp()
+        # self.br = self.registry['business.requirement']
+        self.ModelDataObj = self.env['ir.model.data']
+
+        # Configure unit of measure.
+        self.categ_wtime = self.ModelDataObj.xmlid_to_res_id(
+            'product.uom_categ_wtime')
+        self.categ_kgm = self.ModelDataObj.xmlid_to_res_id(
+            'product.product_uom_categ_kgm')
+        # self.resource = self.env['business.requirement.resource'].create({
```
Comment (duanyp1991) [inline business_requirement_deliverable_cost/tests/test_br.py:19]: remove this

**Comment by cialuo** [comment]
#16

**Pull request closed without merging**